pytroll_collectors/s3stalker.py

```diff
@@ -23,37 +19,133 @@
 
 import logging
 import posixpath
-from datetime import datetime, timedelta
+from datetime import datetime, timedelta, timezone
 import time
+from dateutil import tz
 from contextlib import contextmanager
 import s3fs
-from dateutil import tz
 from posttroll.publisher import Publish
+from posttroll.publisher import create_publisher_from_dict_config
 from trollsift import Parser
 
+import signal
+from threading import Thread
+
 from pytroll_collectors.fsspec_to_message import filelist_unzip_to_messages
 
 logger = logging.getLogger(__name__)
 
 
+class S3StalkerRunner(Thread):
+    """Runner for stalking for new files in an S3 object store."""
+
+    def __init__(self, bucket, config, startup_timedelta_seconds):
+        """Initialize the S3Stalker runner class."""
+        super().__init__()
+
+        self.bucket = bucket
+        self.config = config
+        self.startup_timedelta_seconds = startup_timedelta_seconds
+        self.time_back = self.config['timedelta']
```

Comment:
This is used only in forming `self._wait_seconds` below, so not needed as an attribute.